python313Packages.proton-core: `test_api.py` failed when building package

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

`TestPYSRPClass` in `test_api.py` failed with the error `ValueError: password cannot be longer than 72 bytes`

### Steps to reproduce

`nix build github:nixos/nixpkgs/c5ae371#python313Packages.proton-core`

### Expected behaviour

build tests complete without failures

### Screenshots

_No response_

### Relevant log output

```console
got build log for '/nix/store/3836hmnz6jwdi1mqnmhf7j6c7m5fddhl-python3.13-proton-core-0.4.0.drv' from 'daemon'
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/sdagy0f6g5d2wszpwl1lx0fjw6i0cx25-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/tests/testserver.py"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing pypaBuildPhase
Creating a wheel...
pypa build flags: --no-isolation --outdir dist/ --wheel
[1m* Getting build dependencies for wheel...[0m
/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
!!

        ********************************************************************************
        Please consider removing the following classifiers in favor of a SPDX license expression:

        License :: OSI Approved :: GNU General Public License v3 (GPLv3)

        See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
        ********************************************************************************

!!
  self._finalize_license_expression()
running egg_info
creating proton_core.egg-info
writing proton_core.egg-info/PKG-INFO
writing dependency_links to proton_core.egg-info/dependency_links.txt
writing entry points to proton_core.egg-info/entry_points.txt
writing requirements to proton_core.egg-info/requires.txt
writing top-level names to proton_core.egg-info/top_level.txt
writing manifest file 'proton_core.egg-info/SOURCES.txt'
reading manifest file 'proton_core.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'proton_core.egg-info/SOURCES.txt'
[1m* Building wheel...[0m
/nix/store/5wapciqcq3sqyq19apkxsdb0xl7jhw6b-python3.13-setuptools-80.9.0/lib/python3.13/site-packages/setuptools/dist.py:759: SetuptoolsDeprecationWarning: License classifiers are deprecated.
!!

        ********************************************************************************
        Please consider removing the following classifiers in favor of a SPDX license expression:

        License :: OSI Approved :: GNU General Public License v3 (GPLv3)

        See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
        ********************************************************************************

!!
  self._finalize_license_expression()
running bdist_wheel
running build
running build_py
creating build/lib/proton/keyring
copying proton/keyring/__init__.py -> build/lib/proton/keyring
copying proton/keyring/_base.py -> build/lib/proton/keyring
copying proton/keyring/exceptions.py -> build/lib/proton/keyring
copying proton/keyring/textfile.py -> build/lib/proton/keyring
creating build/lib/proton/loader
copying proton/loader/__init__.py -> build/lib/proton/loader
copying proton/loader/__main__.py -> build/lib/proton/loader
copying proton/loader/loader.py -> build/lib/proton/loader
creating build/lib/proton/session
copying proton/session/__init__.py -> build/lib/proton/session
copying proton/session/api.py -> build/lib/proton/session
copying proton/session/environments.py -> build/lib/proton/session
copying proton/session/exceptions.py -> build/lib/proton/session
copying proton/session/formdata.py -> build/lib/proton/session
creating build/lib/proton/sso
copying proton/sso/__init__.py -> build/lib/proton/sso
copying proton/sso/__main__.py -> build/lib/proton/sso
copying proton/sso/sso.py -> build/lib/proton/sso
creating build/lib/proton/utils
copying proton/utils/__init__.py -> build/lib/proton/utils
copying proton/utils/environment.py -> build/lib/proton/utils
copying proton/utils/metaclasses.py -> build/lib/proton/utils
creating build/lib/proton/views
copying proton/views/__init__.py -> build/lib/proton/views
copying proton/views/_base.py -> build/lib/proton/views
copying proton/views/basiccli.py -> build/lib/proton/views
creating build/lib/proton/session/srp
copying proton/session/srp/__init__.py -> build/lib/proton/session/srp
copying proton/session/srp/_ctsrp.py -> build/lib/proton/session/srp
copying proton/session/srp/_pysrp.py -> build/lib/proton/session/srp
copying proton/session/srp/pmhash.py -> build/lib/proton/session/srp
copying proton/session/srp/util.py -> build/lib/proton/session/srp
creating build/lib/proton/session/transports
copying proton/session/transports/__init__.py -> build/lib/proton/session/transports
copying proton/session/transports/aiohttp.py -> build/lib/proton/session/transports
copying proton/session/transports/alternativerouting.py -> build/lib/proton/session/transports
copying proton/session/transports/auto.py -> build/lib/proton/session/transports
copying proton/session/transports/base.py -> build/lib/proton/session/transports
copying proton/session/transports/requests.py -> build/lib/proton/session/transports
creating build/lib/proton/session/transports/utils
copying proton/session/transports/utils/dns.py -> build/lib/proton/session/transports/utils
running egg_info
writing proton_core.egg-info/PKG-INFO
writing dependency_links to proton_core.egg-info/dependency_links.txt
writing entry points to proton_core.egg-info/entry_points.txt
writing requirements to proton_core.egg-info/requires.txt
writing top-level names to proton_core.egg-info/top_level.txt
reading manifest file 'proton_core.egg-info/SOURCES.txt'
adding license file 'LICENSE'
writing manifest file 'proton_core.egg-info/SOURCES.txt'
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/proton
creating build/bdist.linux-x86_64/wheel/proton/keyring
copying build/lib/proton/keyring/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/keyring
copying build/lib/proton/keyring/_base.py -> build/bdist.linux-x86_64/wheel/./proton/keyring
copying build/lib/proton/keyring/exceptions.py -> build/bdist.linux-x86_64/wheel/./proton/keyring
copying build/lib/proton/keyring/textfile.py -> build/bdist.linux-x86_64/wheel/./proton/keyring
creating build/bdist.linux-x86_64/wheel/proton/loader
copying build/lib/proton/loader/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/loader
copying build/lib/proton/loader/__main__.py -> build/bdist.linux-x86_64/wheel/./proton/loader
copying build/lib/proton/loader/loader.py -> build/bdist.linux-x86_64/wheel/./proton/loader
creating build/bdist.linux-x86_64/wheel/proton/session
copying build/lib/proton/session/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/session
copying build/lib/proton/session/api.py -> build/bdist.linux-x86_64/wheel/./proton/session
copying build/lib/proton/session/environments.py -> build/bdist.linux-x86_64/wheel/./proton/session
copying build/lib/proton/session/exceptions.py -> build/bdist.linux-x86_64/wheel/./proton/session
copying build/lib/proton/session/formdata.py -> build/bdist.linux-x86_64/wheel/./proton/session
creating build/bdist.linux-x86_64/wheel/proton/session/srp
copying build/lib/proton/session/srp/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/session/srp
copying build/lib/proton/session/srp/_ctsrp.py -> build/bdist.linux-x86_64/wheel/./proton/session/srp
copying build/lib/proton/session/srp/_pysrp.py -> build/bdist.linux-x86_64/wheel/./proton/session/srp
copying build/lib/proton/session/srp/pmhash.py -> build/bdist.linux-x86_64/wheel/./proton/session/srp
copying build/lib/proton/session/srp/util.py -> build/bdist.linux-x86_64/wheel/./proton/session/srp
creating build/bdist.linux-x86_64/wheel/proton/session/transports
copying build/lib/proton/session/transports/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/session/transports
copying build/lib/proton/session/transports/aiohttp.py -> build/bdist.linux-x86_64/wheel/./proton/session/transports
copying build/lib/proton/session/transports/alternativerouting.py -> build/bdist.linux-x86_64/wheel/./proton/session/transports
copying build/lib/proton/session/transports/auto.py -> build/bdist.linux-x86_64/wheel/./proton/session/transports
copying build/lib/proton/session/transports/base.py -> build/bdist.linux-x86_64/wheel/./proton/session/transports
copying build/lib/proton/session/transports/requests.py -> build/bdist.linux-x86_64/wheel/./proton/session/transports
creating build/bdist.linux-x86_64/wheel/proton/session/transports/utils
copying build/lib/proton/session/transports/utils/dns.py -> build/bdist.linux-x86_64/wheel/./proton/session/transports/utils
creating build/bdist.linux-x86_64/wheel/proton/sso
copying build/lib/proton/sso/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/sso
copying build/lib/proton/sso/__main__.py -> build/bdist.linux-x86_64/wheel/./proton/sso
copying build/lib/proton/sso/sso.py -> build/bdist.linux-x86_64/wheel/./proton/sso
creating build/bdist.linux-x86_64/wheel/proton/utils
copying build/lib/proton/utils/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/utils
copying build/lib/proton/utils/environment.py -> build/bdist.linux-x86_64/wheel/./proton/utils
copying build/lib/proton/utils/metaclasses.py -> build/bdist.linux-x86_64/wheel/./proton/utils
creating build/bdist.linux-x86_64/wheel/proton/views
copying build/lib/proton/views/__init__.py -> build/bdist.linux-x86_64/wheel/./proton/views
copying build/lib/proton/views/_base.py -> build/bdist.linux-x86_64/wheel/./proton/views
copying build/lib/proton/views/basiccli.py -> build/bdist.linux-x86_64/wheel/./proton/views
running install_egg_info
Copying proton_core.egg-info to build/bdist.linux-x86_64/wheel/./proton_core-0.4.0-py3.13.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/proton_core-0.4.0.dist-info/WHEEL
creating '/build/source/dist/.tmp-y6l0wvjf/proton_core-0.4.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'proton/keyring/__init__.py'
adding 'proton/keyring/_base.py'
adding 'proton/keyring/exceptions.py'
adding 'proton/keyring/textfile.py'
adding 'proton/loader/__init__.py'
adding 'proton/loader/__main__.py'
adding 'proton/loader/loader.py'
adding 'proton/session/__init__.py'
adding 'proton/session/api.py'
adding 'proton/session/environments.py'
adding 'proton/session/exceptions.py'
adding 'proton/session/formdata.py'
adding 'proton/session/srp/__init__.py'
adding 'proton/session/srp/_ctsrp.py'
adding 'proton/session/srp/_pysrp.py'
adding 'proton/session/srp/pmhash.py'
adding 'proton/session/srp/util.py'
adding 'proton/session/transports/__init__.py'
adding 'proton/session/transports/aiohttp.py'
adding 'proton/session/transports/alternativerouting.py'
adding 'proton/session/transports/auto.py'
adding 'proton/session/transports/base.py'
adding 'proton/session/transports/requests.py'
adding 'proton/session/transports/utils/dns.py'
adding 'proton/sso/__init__.py'
adding 'proton/sso/__main__.py'
adding 'proton/sso/sso.py'
adding 'proton/utils/__init__.py'
adding 'proton/utils/environment.py'
adding 'proton/utils/metaclasses.py'
adding 'proton/views/__init__.py'
adding 'proton/views/_base.py'
adding 'proton/views/basiccli.py'
adding 'proton_core-0.4.0.dist-info/licenses/LICENSE'
adding 'proton_core-0.4.0.dist-info/METADATA'
adding 'proton_core-0.4.0.dist-info/WHEEL'
adding 'proton_core-0.4.0.dist-info/entry_points.txt'
adding 'proton_core-0.4.0.dist-info/top_level.txt'
adding 'proton_core-0.4.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
[1m[92mSuccessfully built [4mproton_core-0.4.0-py3-none-any.whl[0m[1m[92m[0m
Finished creating a wheel...
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { "action": "setPhase", "phase": "pythonRuntimeDepsCheckHook" }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for proton_core-0.4.0-py3-none-any.whl
Finished executing pythonRuntimeDepsCheck
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Executing pypaInstallPhase
Successfully installed proton_core-0.4.0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
@nix { "action": "setPhase", "phase": "pythonOutputDistPhase" }
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/w6mg299az3in9gm4xrrk1c28wa9zbpw5-python3.13-proton-core-0.4.0
checking for references to /build/ in /nix/store/w6mg299az3in9gm4xrrk1c28wa9zbpw5-python3.13-proton-core-0.4.0...
patching script interpreter paths in /nix/store/w6mg299az3in9gm4xrrk1c28wa9zbpw5-python3.13-proton-core-0.4.0
stripping (with command strip and flags -S -p) in  /nix/store/w6mg299az3in9gm4xrrk1c28wa9zbpw5-python3.13-proton-core-0.4.0/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/mp9xhf8pwrc496sgz5p425wlcr3mmlqw-python3.13-proton-core-0.4.0-dist
checking for references to /build/ in /nix/store/mp9xhf8pwrc496sgz5p425wlcr3mmlqw-python3.13-proton-core-0.4.0-dist...
patching script interpreter paths in /nix/store/mp9xhf8pwrc496sgz5p425wlcr3mmlqw-python3.13-proton-core-0.4.0-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
@nix { "action": "setPhase", "phase": "pythonCatchConflictsPhase" }
Running phase: pythonRemoveBinBytecodePhase
@nix { "action": "setPhase", "phase": "pythonRemoveBinBytecodePhase" }
Running phase: pythonImportsCheckPhase
@nix { "action": "setPhase", "phase": "pythonImportsCheckPhase" }
Executing pythonImportsCheckPhase
Check whether the following modules can be imported: proton
Running phase: pytestCheckPhase
@nix { "action": "setPhase", "phase": "pytestCheckPhase" }
Executing pytestCheckPhase
pytest flags: -m pytest --ignore-glob=tests/test_alternativerouting.py -k not\ \(test_modulus_verification\)\ and\ not\ \(test_broken_data\)\ and\ not\ \(test_broken_index\)\ and\ not\ \(test_sessions\)\ and\ not\ \(test_auto_works_on_prod\)\ and\ not\ \(test_ping\)\ and\ not\ \(test_successful\)\ and\ not\ \(test_without_pinning\)\ and\ not\ \(test_bad_pinning_fingerprint_changed\)\ and\ not\ \(test_bad_pinning_url_changed\)
[1m============================= test session starts ==============================[0m
platform linux -- Python 3.13.9, pytest-8.4.2, pluggy-1.6.0
rootdir: /build/source
configfile: setup.cfg
testpaths: tests
plugins: cov-nixpkgs-stub-1.0.0
[1mcollecting ... [0m[1m
collecting 107 items                                                           [0m[1m
collected 124 items / 11 deselected / 113 selected                             [0m

tests/test_aiohttp_transport.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m                                     [  3%][0m
tests/test_api.py [33ms[0m[33ms[0m[33ms[0m[33ms[0m[31mF[0m[31mF[0m[32m.[0m[31mF[0m[31m                                               [ 10%][0m
tests/test_autotransport.py [32m.[0m[32m.[0m[31m                                           [ 12%][0m
tests/test_basekeyring.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m [ 53%]
[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                                                    [ 57%][0m
tests/test_dns_requests.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                      [ 78%][0m
tests/test_environment.py [32m.[0m[31m                                              [ 79%][0m
tests/test_loader.py [32m.[0m[32m.[0m[32m.[0m[31m                                                 [ 82%][0m
tests/test_protonsso.py [33ms[0m[33ms[0m[33ms[0m[31m                                              [ 84%][0m
tests/test_requests_transport.py [32m.[0m[32m.[0m[32m.[0m[31m                                     [ 87%][0m
tests/test_session.py [32m.[0m[33ms[0m[33ms[0m[31m                                                [ 90%][0m
tests/test_session_pickle.py [32m.[0m[31m                                           [ 91%][0m
tests/test_textfilekeyring.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                   [ 98%][0m
tests/test_tlsverification.py [32m.[0m[32m.[0m[31m                                         [100%][0m

=================================== FAILURES ===================================
[31m[1m________________________ TestPYSRPClass.test_compute_v _________________________[0m

self = <test_api.TestPYSRPClass testMethod=test_compute_v>

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mtest_compute_v[39;49;00m([96mself[39;49;00m):[90m[39;49;00m
        [94mfor[39;49;00m instance [95min[39;49;00m srp_instances:[90m[39;49;00m
            [94mif[39;49;00m instance[[33m"[39;49;00m[33mException[39;49;00m[33m"[39;49;00m] [95mis[39;49;00m [95mnot[39;49;00m [94mNone[39;49;00m:[90m[39;49;00m
                [94mwith[39;49;00m [96mself[39;49;00m.assertRaises(instance[[33m'[39;49;00m[33mException[39;49;00m[33m'[39;49;00m]):[90m[39;49;00m
                    usr = [96mself[39;49;00m.user([90m[39;49;00m
                        instance[[33m"[39;49;00m[33mPassword[39;49;00m[33m"[39;49;00m],[90m[39;49;00m
                        [96mbytes[39;49;00m.fromhex(instance[[33m"[39;49;00m[33mModulus[39;49;00m[33m"[39;49;00m])[90m[39;49;00m
                    )[90m[39;49;00m
                    usr.compute_v([90m[39;49;00m
                        base64.b64decode(instance[[33m"[39;49;00m[33mSalt[39;49;00m[33m"[39;49;00m]), PM_VERSION[90m[39;49;00m
                    )[90m[39;49;00m
            [94melse[39;49;00m:[90m[39;49;00m
                usr = [96mself[39;49;00m.user([90m[39;49;00m
                    instance[[33m"[39;49;00m[33mPassword[39;49;00m[33m"[39;49;00m],[90m[39;49;00m
                    [96mbytes[39;49;00m.fromhex(instance[[33m"[39;49;00m[33mModulus[39;49;00m[33m"[39;49;00m])[90m[39;49;00m
                )[90m[39;49;00m
>               salt, v = usr.compute_v([90m[39;49;00m
                    base64.b64decode(instance[[33m"[39;49;00m[33mSalt[39;49;00m[33m"[39;49;00m]), PM_VERSION[90m[39;49;00m
                )[90m[39;49;00m

[1m[31mtests/test_api.py[0m:59: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[1m[31mproton/session/srp/_pysrp.py[0m:158: in compute_v
    [0m[96mself[39;49;00m.x = calculate_x([96mself[39;49;00m.hash_class, [96mself[39;49;00m.bytes_s, [96mself[39;49;00m.p, [96mself[39;49;00m.N, version)[90m[39;49;00m
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
[1m[31mproton/session/srp/_pysrp.py[0m:50: in calculate_x
    [0mexp = hash_password([90m[39;49;00m
[1m[31mproton/session/srp/util.py[0m:48: in hash_password
    [0m[94mreturn[39;49;00m hash_password_3(hash_class, password, salt, modulus)[90m[39;49;00m
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

hash_class = <function pmhash at 0x7ffff5995d00>
password = b'PasswordThatIsLongerThan72CharsAndContainsSomeRandomStuffThatNoOneCaresAbout'
salt = b'jQGE9ZrERiIQ2V/wZ1PtZe'
modulus = b'\xc3v7z\xc6\xc6&\x97\xe2\x11\xc0\xbe\x80\xda<_{s\x81\xab\x92\xd9E6\xb4\x06YJ\xd0\xc1\xbf\xe9\nBN6\x08\\\xd2\xf5S\xf3...xf7!\xeb\xf1\xd9\x10[\xe3<\xdc\x19\xa0\x10\x80\xa9B\x7fOv\xb2}?\xcf\x89&\xa5\xc4\x88LB\xd1\xb6\xd0R\xc2\xf7DE"\x83\xdb'

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mhash_password_3[39;49;00m(hash_class, password, salt, modulus):[90m[39;49;00m
        salt = (salt + [33mb[39;49;00m[33m"[39;49;00m[33mproton[39;49;00m[33m"[39;49;00m)[:[94m16[39;49;00m][90m[39;49;00m
        salt = bcrypt_b64_encode(salt)[:[94m22[39;49;00m][90m[39;49;00m
>       hashed = bcrypt.hashpw(password, [33mb[39;49;00m[33m"[39;49;00m[33m$2y$10$[39;49;00m[33m"[39;49;00m + salt)[90m[39;49;00m
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
[1m[31mE       ValueError: password cannot be longer than 72 bytes, truncate manually if necessary (e.g. my_password[:72])[0m

[1m[31mproton/session/srp/util.py[0m:42: ValueError
[31m[1m________________________ TestPYSRPClass.test_generate_v ________________________[0m

self = <test_api.TestPYSRPClass testMethod=test_generate_v>

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mtest_generate_v[39;49;00m([96mself[39;49;00m):[90m[39;49;00m
        [94mfor[39;49;00m instance [95min[39;49;00m srp_instances:[90m[39;49;00m
            [94mif[39;49;00m instance[[33m"[39;49;00m[33mException[39;49;00m[33m"[39;49;00m] [95mis[39;49;00m [95mnot[39;49;00m [94mNone[39;49;00m:[90m[39;49;00m
                [94mcontinue[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
            usr = [96mself[39;49;00m.user([90m[39;49;00m
                instance[[33m"[39;49;00m[33mPassword[39;49;00m[33m"[39;49;00m],[90m[39;49;00m
                [96mbytes[39;49;00m.fromhex(instance[[33m"[39;49;00m[33mModulus[39;49;00m[33m"[39;49;00m])[90m[39;49;00m
            )[90m[39;49;00m
>           generated_salt, generated_v = usr.compute_v()[90m[39;49;00m
                                          ^^^^^^^^^^^^^^^[90m[39;49;00m

[1m[31mtests/test_api.py[0m:86: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[1m[31mproton/session/srp/_pysrp.py[0m:158: in compute_v
    [0m[96mself[39;49;00m.x = calculate_x([96mself[39;49;00m.hash_class, [96mself[39;49;00m.bytes_s, [96mself[39;49;00m.p, [96mself[39;49;00m.N, version)[90m[39;49;00m
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
[1m[31mproton/session/srp/_pysrp.py[0m:50: in calculate_x
    [0mexp = hash_password([90m[39;49;00m
[1m[31mproton/session/srp/util.py[0m:48: in hash_password
    [0m[94mreturn[39;49;00m hash_password_3(hash_class, password, salt, modulus)[90m[39;49;00m
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

hash_class = <function pmhash at 0x7ffff5995d00>
password = b'PasswordThatIsLongerThan72CharsAndContainsSomeRandomStuffThatNoOneCaresAbout'
salt = b'PuivYmgG8Ezp0V/wZ1PtZe'
modulus = b'\xc3v7z\xc6\xc6&\x97\xe2\x11\xc0\xbe\x80\xda<_{s\x81\xab\x92\xd9E6\xb4\x06YJ\xd0\xc1\xbf\xe9\nBN6\x08\\\xd2\xf5S\xf3...xf7!\xeb\xf1\xd9\x10[\xe3<\xdc\x19\xa0\x10\x80\xa9B\x7fOv\xb2}?\xcf\x89&\xa5\xc4\x88LB\xd1\xb6\xd0R\xc2\xf7DE"\x83\xdb'

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mhash_password_3[39;49;00m(hash_class, password, salt, modulus):[90m[39;49;00m
        salt = (salt + [33mb[39;49;00m[33m"[39;49;00m[33mproton[39;49;00m[33m"[39;49;00m)[:[94m16[39;49;00m][90m[39;49;00m
        salt = bcrypt_b64_encode(salt)[:[94m22[39;49;00m][90m[39;49;00m
>       hashed = bcrypt.hashpw(password, [33mb[39;49;00m[33m"[39;49;00m[33m$2y$10$[39;49;00m[33m"[39;49;00m + salt)[90m[39;49;00m
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
[1m[31mE       ValueError: password cannot be longer than 72 bytes, truncate manually if necessary (e.g. my_password[:72])[0m

[1m[31mproton/session/srp/util.py[0m:42: ValueError
[31m[1m___________________________ TestPYSRPClass.test_srp ____________________________[0m

self = <test_api.TestPYSRPClass testMethod=test_srp>

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mtest_srp[39;49;00m([96mself[39;49;00m):[90m[39;49;00m
        [94mfor[39;49;00m instance [95min[39;49;00m srp_instances:[90m[39;49;00m
            [94mif[39;49;00m instance[[33m"[39;49;00m[33mException[39;49;00m[33m"[39;49;00m]:[90m[39;49;00m
                [94mcontinue[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
            server = TestServer()[90m[39;49;00m
    [90m[39;49;00m
            server.setup([90m[39;49;00m
                instance[[33m"[39;49;00m[33mUsername[39;49;00m[33m"[39;49;00m],[90m[39;49;00m
                [96mbytes[39;49;00m.fromhex(instance[[33m"[39;49;00m[33mModulus[39;49;00m[33m"[39;49;00m]),[90m[39;49;00m
                base64.b64decode(instance[[33m"[39;49;00m[33mVerifier[39;49;00m[33m"[39;49;00m])[90m[39;49;00m
            )[90m[39;49;00m
    [90m[39;49;00m
            server_challenge = server.get_challenge()[90m[39;49;00m
            usr = [96mself[39;49;00m.user([90m[39;49;00m
                instance[[33m"[39;49;00m[33mPassword[39;49;00m[33m"[39;49;00m], [96mbytes[39;49;00m.fromhex(instance[[33m"[39;49;00m[33mModulus[39;49;00m[33m"[39;49;00m])[90m[39;49;00m
            )[90m[39;49;00m
    [90m[39;49;00m
            client_challenge = usr.get_challenge()[90m[39;49;00m
>           client_proof = usr.process_challenge([90m[39;49;00m
                base64.b64decode(instance[[33m"[39;49;00m[33mSalt[39;49;00m[33m"[39;49;00m]),[90m[39;49;00m
                server_challenge,[90m[39;49;00m
                PM_VERSION[90m[39;49;00m
            )[90m[39;49;00m

[1m[31mtests/test_api.py[0m:123: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[1m[31mproton/session/srp/_pysrp.py[0m:136: in process_challenge
    [0m[96mself[39;49;00m.x = calculate_x([96mself[39;49;00m.hash_class, [96mself[39;49;00m.bytes_s, [96mself[39;49;00m.p, [96mself[39;49;00m.N, version)[90m[39;49;00m
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
[1m[31mproton/session/srp/_pysrp.py[0m:50: in calculate_x
    [0mexp = hash_password([90m[39;49;00m
[1m[31mproton/session/srp/util.py[0m:48: in hash_password
    [0m[94mreturn[39;49;00m hash_password_3(hash_class, password, salt, modulus)[90m[39;49;00m
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

hash_class = <function pmhash at 0x7ffff5995d00>
password = b'PasswordThatIsLongerThan72CharsAndContainsSomeRandomStuffThatNoOneCaresAbout'
salt = b'jQGE9ZrERiIQ2V/wZ1PtZe'
modulus = b'\xc3v7z\xc6\xc6&\x97\xe2\x11\xc0\xbe\x80\xda<_{s\x81\xab\x92\xd9E6\xb4\x06YJ\xd0\xc1\xbf\xe9\nBN6\x08\\\xd2\xf5S\xf3...xf7!\xeb\xf1\xd9\x10[\xe3<\xdc\x19\xa0\x10\x80\xa9B\x7fOv\xb2}?\xcf\x89&\xa5\xc4\x88LB\xd1\xb6\xd0R\xc2\xf7DE"\x83\xdb'

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mhash_password_3[39;49;00m(hash_class, password, salt, modulus):[90m[39;49;00m
        salt = (salt + [33mb[39;49;00m[33m"[39;49;00m[33mproton[39;49;00m[33m"[39;49;00m)[:[94m16[39;49;00m][90m[39;49;00m
        salt = bcrypt_b64_encode(salt)[:[94m22[39;49;00m][90m[39;49;00m
>       hashed = bcrypt.hashpw(password, [33mb[39;49;00m[33m"[39;49;00m[33m$2y$10$[39;49;00m[33m"[39;49;00m + salt)[90m[39;49;00m
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^[90m[39;49;00m
[1m[31mE       ValueError: password cannot be longer than 72 bytes, truncate manually if necessary (e.g. my_password[:72])[0m

[1m[31mproton/session/srp/util.py[0m:42: ValueError
[36m[1m=========================== short test summary info ============================[0m
[31mFAILED[0m tests/test_api.py::[1mTestPYSRPClass::test_compute_v[0m - ValueError: password cannot be longer than 72 bytes, truncate manually if n...
[31mFAILED[0m tests/test_api.py::[1mTestPYSRPClass::test_generate_v[0m - ValueError: password cannot be longer than 72 bytes, truncate manually if n...
[31mFAILED[0m tests/test_api.py::[1mTestPYSRPClass::test_srp[0m - ValueError: password cannot be longer than 72 bytes, truncate manually if n...
[31m=========== [31m[1m3 failed[0m, [32m101 passed[0m, [33m9 skipped[0m, [33m11 deselected[0m[31m in 15.85s[0m[31m ===========[0m
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.7, NixOS, 25.11 (Xantusia), 25.11.20251111.9da7f1c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.33pre20251107_479b6b73`
 - nixpkgs: `/nix/store/cjjpqrp9xy44j5fjh3mygyxy3i7mslj2-source`

### Notify maintainers

@SebTM 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+-label%3A%226.topic%3A+darwin%22+-label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/proton-core/default.nix
+++ b/pkgs/development/python-modules/proton-core/default.nix
@@ -63,6 +63,11 @@ buildPythonPackage rec {
     # Failed assertions
     "test_bad_pinning_fingerprint_changed"
     "test_bad_pinning_url_changed"
+    # Bcrypt 72-byte limit exceeded
+    # https://github.com/ProtonVPN/python-proton-core/pull/10
+    "test_compute_v"
+    "test_generate_v"
+    "test_srp"
   ];
 
   meta = {

